highfidelity/hifi
# 8027 QML & Web generated audio now follow the client audio device selection

This PR makes QML & Web surfaces responsive to changes on the client audio selection. 

https://highfidelity.fogbugz.com/f/cases/8027

...

## TEST PLAN
== WEB ENTITIES
- Go to the Sandbox or other domain where you can create entities.
- Create a new web entity.
- Change its url to youtube.com.
- Select any video and start watching ( they should reproduce sound using the current device ).
- Go to the audio menu and change the output audio device.
- The video will reproduce sound using the selected device.
- Delete the web entity.

== WEB BROWSER
- Open the interface web browser (CTRL + SHIFT + B).
- Change its url to youtube.com.
- Select any video and start watching ( they should reproduce sound using the current device ).
- Go to the audio menu and change the output audio device.
- The video will reproduce sound using the selected device.
- Close the browser.

== QML MULTIMEDIA
- Open the script window (Edit -> Running Scripts... or CTRL + J) 
- Click on "FROM URL" and add the link https://s3-us-west-1.amazonaws.com/hifio/qmlmediatest.js
- An audio loaded from a QML file should start playing using the selected output device. 
- Go to the audio menu and change the output audio device.
- The audio will reproduce sound using the selected device.

...

## KNOWN ISSUES
- Only media DOM elements will be updated. Embedded media elements won’t (youtube videos on hifidelity.io).
- Every time a HTML media elements gets added to the DOM, the default audio output device is automatically asign as their output audio. This means that every time the page loads or changes its content, all the DOM elements will need to be updated with the selected audio device. In some cases this changes are difficult or impossible to cast.( a <video> is added 3 seconds after page loads ) 
- QML video only accepts the wmv file format and doesn't render propertly (only audio).
